src/common/utils/storage.ts

```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+
+// a single place to define all used localStorage keys, to avoid collisions or hardcoded strings
+export enum StorageKeys {
+  COUNTY_MAP_TYPE = 'COUNTY_MAP_TYPE',
+  FULLSTORY_RECORD = 'FULLSTORY_RECORD',
+}
+
+// From https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#testing_for_availability
+export function storageAvailable(storageType: string) {
+  let storage;
+  const testKey = '__storage_test__';
+  try {
+    // @ts-ignore
+    storage = window[storageType];
+    storage.setItem(testKey, testKey);
+    storage.removeItem(testKey);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+// Hook from https://usehooks.com/useLocalStorage/
+// but updated to fallback to setState if localstorage is not available.
+export function useLocalStorage<T>(
+  key: StorageKeys,
+  initialValue: T,
+): [T, (value: T) => void] {
+  // State to store our value
+  // Pass initial state function to useState so logic is only executed once
+  const [storedValue, setStoredValue] = useState(() => {
+    try {
+      // Get from local storage by key
+      let item;
+      if (storageAvailable('localStorage')) {
```

Comment:
the funny thing is that there's already a try/catch around this so the `storageAvailable` call really needed? (other than signaling a little more strongly to the developer?)